File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None: return None
        curr, prev, nxt = head, None, None
        cnt = 0
        while curr != None and cnt < 2:
            nxt = curr.next
            curr.next = prev
            prev = curr
            curr = nxt
            cnt += 1
        head.next = self.swapPairs(nxt)
        return prev

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null) return null;
        ListNode curr = head, prev = null, nxt = null;
        int cnt = 0;
        while(curr!=null && cnt<2){
            nxt = curr.next;
            curr.next = prev;
            prev = curr;
            curr = nxt;
            cnt++;
        }
        head.next = swapPairs(nxt);
        return prev;
    }
}